hexa/pipeline_templates/models.py

```diff
@@ -77,6 +78,20 @@ def upgrade_pipeline(
             principal, pipeline.workspace, pipeline
         )
 
+    def new_versions(self, pipeline: Pipeline) -> list["PipelineTemplateVersion"]:
+        """Return the versions of the template that are newer than the last version of the pipeline"""
+        if pipeline.source_template != self:
+            raise ValueError("The specified pipeline is not from this template")
+
+        last_version_from_template = pipeline.versions.filter(
```

Comment:
nitpicking: to ease my understanding of the clause, I would have use this wording. I do not think it's worth it to change it. 
```suggestion
        last_version_created_from_template = pipeline.versions.filter(
```